identity-core/src/utils/base_encoding.rs

```diff
@@ -67,52 +67,73 @@ impl From<Base> for multibase::Base {
   }
 }
 
-/// Decodes the given `data` as [Multibase] with an inferred [`base`](Base).
-///
-/// [Multibase]: https://datatracker.ietf.org/doc/html/draft-multiformats-multibase-03
-pub fn decode_multibase<T>(data: &T) -> Result<Vec<u8>>
-where
-  T: AsRef<str> + ?Sized,
-{
-  if data.as_ref().is_empty() {
-    return Ok(Vec::new());
+/// Provides utility functions for encoding and decoding between various bases.
+pub struct BaseEncoding;
+
+impl BaseEncoding {
+  /// Encodes the given `data` to the specified [`base`](Base).
+  pub fn encode<T>(data: &T, base: Base) -> String
+  where
+    T: AsRef<[u8]> + ?Sized,
+  {
+    multibase::Base::from(base).encode(data)
   }
-  multibase::decode(&data)
-    .map(|(_base, output)| output)
-    .map_err(Error::DecodeMultibase)
-}
 
-/// Encodes the given `data` as [Multibase] with the given [`base`](Base), defaults to
-/// [`Base::Base58Btc`] if omitted.
-///
-/// NOTE: [`encode_multibase`] with [`Base::Base58Btc`] is different from [`encode_b58`] as
-/// the [Multibase] format prepends a base-encoding-character to the output.
-///
-/// [Multibase]: https://datatracker.ietf.org/doc/html/draft-multiformats-multibase-03
-pub fn encode_multibase<T>(data: &T, base: Option<Base>) -> String
-where
-  T: AsRef<[u8]> + ?Sized,
-{
-  multibase::encode(multibase::Base::from(base.unwrap_or(Base::Base58Btc)), data)
-}
+  /// Decodes the given `data` encoded as the specified [`base`](Base).
+  pub fn decode<T>(data: &T, base: Base) -> Result<Vec<u8>>
+  where
+    T: AsRef<str> + ?Sized,
+  {
+    multibase::Base::from(base)
+      .decode(data)
+      .map_err(|err| Error::DecodeBase(base, err))
+  }
 
-/// Decodes the given `data` as base58-btc.
-pub fn decode_b58<T>(data: &T) -> Result<Vec<u8>>
-where
-  T: AsRef<[u8]> + ?Sized,
-{
-  bs58::decode(data)
-    .with_alphabet(bs58::Alphabet::BITCOIN)
-    .into_vec()
-    .map_err(Error::DecodeBase58)
-}
+  /// Encodes the given `data` to [`Base::Base58Btc`].
+  pub fn encode_base58<T>(data: &T) -> String
```

Comment:
Is this a convenience method for `|data| Self::encode(data, Base::Base58Btc`)`? 

If so it might be an idea to state that the two are equivalent in the documentation.